server/routes/tasks.js

```diff
@@ -1,8 +1,39 @@
+const formalizeMultiselectValues = (values) => [values].flat()
+  .filter((it) => !!it)
+  .map((value) => Number(value));
+
 export default (app) => {
   app
     .get('/tasks', async (req, reply) => {
-      const tasks = await app.objection.models.task.query().withGraphJoined('[status, creator, assigned, labels]');
-      await reply.render('tasks/index', { data: { tasks } });
+      const statusIds = formalizeMultiselectValues(req.query.statusIds);
+      const labelIds = formalizeMultiselectValues(req.query.labelIds);
+      const assignedIds = formalizeMultiselectValues(req.query.assignedIds);
+
+      const [filteredTasks, statuses, labels, users] = await Promise.all([
+        (async () => {
+          let tasks = app.objection.models.task.query();
+          if (statusIds.length) {
+            tasks = tasks.modify('withStatusIn', statusIds);
+          }
+          if (assignedIds.length) {
+            tasks = tasks.modify('withAssignedIn', assignedIds);
+          }
+
+          tasks = tasks.withGraphJoined('[status, creator, assigned, labels]');
+          if (labelIds.length) {
+            tasks = tasks.modify('withLabelIn', labelIds);
+          }
+          return tasks;
+        })(),
```

Comment:
query is an instance of https://knexjs.org/. It's mutable

So you can go like:
const tasksQuery = app.objection.models.task.query();
if(status.length) tasksQuery.modify('withStatusIn', statusIds)
....

const tasks = await tasksQuery;

Reply:
hm, why would we need to wrap query building in anonymus async function? lets move it above promise.all